bimsb/packages/bioinformatics-nonfree.scm

```diff
@@ -1882,3 +1883,95 @@ MACE is a bioinformatics tool dedicated to analyze ChIP-exo data.")
+             (let ((out (assoc-ref outputs "out")))
+               ;; patch scripts checking for ../install_successful file
+               (for-each
+                 (lambda (script)
+                   (substitute* script
+                     (("\\$bn/../install_successful")
+                      (string-append out
+                                     "/share/mirdeep2/install_successful"))))
+                 '("src/mapper.pl"
+                   "src/miRDeep2.pl"
+                   "src/quantifier.pl"))
+               ;; patch script using ../Rfam_for_miRDeep.fa file
+               (substitute* "src/miRDeep2.pl"
+                 (("\\$\\{scripts\\}/../Rfam_for_miRDeep.fa")
+                  (string-append out "/share/mirdeep2/Rfam_for_miRDeep.fa"))
+                 (("\\$scripts/../Rfam_for_miRDeep.fa")
+                  (string-append out "/share/mirdeep2/Rfam_for_miRDeep.fa"))))
+             #t))
+         (replace 'install
+           (lambda* (#:key inputs outputs #:allow-other-keys)
+             (let ((out (assoc-ref outputs "out")))
+               (with-directory-excursion "src"
+                 (copy-recursively "." (string-append out "/bin")))
+               ;; place Rfam_for_miRDeep.fa in /share/mirdeep2
+               (mkdir-p (string-append out "/share/mirdeep2"))
+               (copy-file
+                 "Rfam_for_miRDeep.fa"
+                 (string-append out "/share/mirdeep2/Rfam_for_miRDeep.fa"))
+               ;; create install_successful file in share/mirdeep2
+               (with-output-to-file
+                 (string-append out "/share/mirdeep2/install_successful")
+                 (const #t)))  ;; simply touch the file
+             #t))
+         (add-after 'install 'wrap-perl-scripts
+          (lambda* (#:key inputs outputs #:allow-other-keys)
+            (let ((out (assoc-ref outputs "out")))
+              ;; Make sure perl scripts find all perl inputs at runtime.
+              (for-each (lambda (prog)
+                          (wrap-program (string-append out "/bin/" prog)
+                            `("PERL5LIB" ":" prefix
+                              (,(getenv "PERL5LIB")))))
+                        '("make_html2.pl"
+                          "make_html.pl"
+                          "miRDeep2.pl"))
+              ;; Make sure perl scripts find all input binaries at runtime.
+              (for-each (lambda (prog)
+                          (wrap-program (string-append out "/bin/" prog)
+                            `("PATH" ":" prefix
+                               (,(getenv "PATH")))))
+                        '("make_html2.pl"
+                          "make_html.pl"
+                          "mapper.pl"
+                          "miRDeep2_core_algorithm.pl"
+                          "miRDeep2.pl"
+                          "prepare_signature.pl"
+                          "quantifier.pl"))
+              #t))))))
+    (inputs
+      `(("bowtie1" ,bowtie1)
```

Comment:
While I can use miRDeep2 without being able to run `bowtie` directly (thx to the wrapping), I still need a mapping index. So in most cases, I'll need to run `bowtie-build` (provided by `bowtie1` package) before I can run miRDeep2. So as a user I would be surprised that I have to `guix package -i bowtie1` even though I already did `guix package -i mirdeep2`.
Is that the way it should be (as strictly I *can* run miRDeep2 if I got an indexed genome elsewhere) or is this a case where the input should be propagated?